src/main/java/com/redhat/devtools/lsp4ij/features/completion/LSPCompletionContributor.java

```diff
@@ -136,11 +143,23 @@ private void addCompletionItems(@NotNull CompletionParameters parameters,
             // Create lookup item
             LookupElement lookupItem = completionFeature.createLookupElement(item, context);
             if (lookupItem != null) {
-               completionFeature.addLookupItem(completionPrefix, result, lookupItem, size- i, item);
+                completionFeature.addLookupItem(completionPrefix, result, lookupItem, size - i, item, caseSensitive);
             }
         }
     }
 
+    private static boolean isCaseSensitive(@NotNull LanguageServerItem languageServer, @NotNull PsiFile file) {
```

Comment:
This should be pretty straightforward, but basically try to infer language grammar case-sensitivity via the file => language ID => client-side settings.